### PR TITLE
Avoid sorting in more situations.

### PR DESCRIPTION
**Pull request by petermattis**
Consider the query:

  SELECT v FROM t WHERE k = 1 ORDER BY v

If there is an index on (k, v) we know that an exact match has been
performed on k and thus the results are already ordered on v and thus no
sorting is required.

Added an additional return value to planNode.Ordering() to indicate the
prefix of the ordering for which an exact match has been performed. A
bit of code reorg allowed sortNode.wrap() to take advantage of this new
info.

See #2636.

### CODE REVIEW

File: sql/select.go [hunk 1]
```diff
@@ -285,50 +286,17 @@ func (v *indexInfo) analyzeRanges(exprs []parser.Exprs) {
...
 	// prefix is inconsequential for ordering because the values are identical.
-	prefix := exactPrefix(v.constraints)
+	v.exactPrefix = exactPrefix(v.constraints)
+
+	// Compute the ordering provided by the index.
+	indexOrdering := scan.computeOrdering(v.index.fullColumnIDs())
 
 	// Compute how much of the index ordering matches the requested ordering for
 	// both forward and reverse scans.
-	fwdMatch, fwdPrefix, fwdIndexOrdering := 0, prefix, indexOrdering
```
Comment (tamird) [inline sql/select.go]: this is an unrelated refactor, yeah?

**Comment by tamird** [comment]
LGTM

File: sql/select.go (hunk 1 above)
Comment (petermattis) [inline sql/select.go]: Not quite. The `computeOrderingMatch` this was refactored into is also used by `sortNode.wrap`.
